sklearn/model_selection/tests/test_search.py

```diff
@@ -133,12 +133,12 @@ def assert_grid_iter_equals_getitem(grid):
 @pytest.mark.parametrize(
     "input, error_type, error_message",
     [
-        (0, TypeError, r"Parameter .* is not a dict or a list \(0\)"),
+        (0, TypeError, r"Parameter .* a dict or a list, got: 0 of type .*"),
         ([{"foo": [0]}, 0], TypeError, r"Parameter .* is not a dict \(0\)"),
         (
             {"foo": 0},
             TypeError,
-            "Parameter.* value is not iterable .*" r"\(key='foo', value=0\)",
+            r"Parameter .*",
```

Comment:
> @ogrisel Updated the error message to give more info as well as in the third test for test_validate_parameter_input, there is very less similarity between parameter sampler and parameter grid so added a .*

Please at least check that the parameter name is mentioned, and ideally more details such as:

```
r"Parameter (grid|distribution) for parameter 'foo' is not (a list or an array|iterable or a distribution).*"
```

That will require adjusting the message raised by `ParameterSampler` to use a consistent structure.

Reply:
Sorry didn't knew how to use ( | ) in regex. Thank for the great help.

Reply:
I believe this point was not fully addressed because the pattern is still `r"Parameter .*"`. I think it should at least look for the parameter name.

Reply:
Done